django/db/backends/utils.py

```diff
@@ -114,6 +114,96 @@ def _executemany(self, sql, param_list, *ignored_wrapper_args):
             return self.cursor.executemany(sql, param_list)
 
 
+class AsyncCursorCtx:
+    """
+    Asynchronous context manager to hold an async cursor.
+    """
+
+    def __init__(self, db, name=None):
+        self.db = db
+        self.name = name
+        self.wrap_database_errors = self.db.wrap_database_errors
+
+    async def __aenter__(self):
+        await self.db.aclose_if_health_check_failed()
+        await self.db.aensure_connection()
+        self.wrap_database_errors.__enter__()
+        return self.db._aprepare_cursor(self.db.create_async_cursor(self.name))
+
+    async def __aexit__(self, type, value, traceback):
+        self.wrap_database_errors.__exit__(type, value, traceback)
+
+
+class AsyncCursorWrapper(CursorWrapper):
+    async def _aexecute(self, sql, params, *ignored_wrapper_args):
+        # Raise a warning during app initialization (stored_app_configs is only
+        # ever set during testing).
+        if not apps.ready and not apps.stored_app_configs:
+            warnings.warn(self.APPS_NOT_READY_WARNING_MSG, category=RuntimeWarning)
+        self.db.validate_no_broken_transaction()
+        with self.db.wrap_database_errors:
+            if params is None:
+                # params default might be backend specific.
+                return await self.cursor.execute(sql)
+            else:
+                return await self.cursor.execute(sql, params)
+
+    async def _aexecute_with_wrappers(self, sql, params, many, executor):
+        context = {"connection": self.db, "cursor": self}
+        for wrapper in reversed(self.db.execute_wrappers):
+            executor = functools.partial(wrapper, executor)
+        return await executor(sql, params, many, context)
+
+    async def aexecute(self, sql, params=None):
+        return await self._aexecute_with_wrappers(
+            sql, params, many=False, executor=self._aexecute
+        )
+
+    async def _aexecutemany(self, sql, param_list, *ignored_wrapper_args):
+        # Raise a warning during app initialization (stored_app_configs is only
+        # ever set during testing).
+        if not apps.ready and not apps.stored_app_configs:
+            warnings.warn(self.APPS_NOT_READY_WARNING_MSG, category=RuntimeWarning)
+        self.db.validate_no_broken_transaction()
+        with self.db.wrap_database_errors:
+            return await self.cursor.executemany(sql, param_list)
+
+    async def aexecutemany(self, sql, param_list):
+        return await self._aexecute_with_wrappers(
+            sql, param_list, many=True, executor=self._aexecutemany
+        )
+
+    async def afetchone(self, *args, **kwargs):
```

Comment:
It looks like we need to add `self.db.wrap_database_errors` manager for each method 

- afetchone
- afetchall
- astream
- ascroll
- `__aiter__` + yield from